types/block.go

```diff
@@ -1400,10 +1399,32 @@ func (data *Data) ComputeShares() (NamespacedShares, int) {
 		tailShares...), curLen
 }
 
-func getNextSquareNum(length int) int {
-	width := int(math.Ceil(math.Sqrt(float64(length))))
-	// TODO(ismail): make width a power of two instead
-	return width * width
+// paddedLen calculates the number of shares needed to make a power of 2 square
+// given the current number of shares
+func paddedLen(length int) int {
+	width := uint32(math.Ceil(math.Sqrt(float64(length))))
+	width = nextPowerOf2(width)
+	return int(width * width)
+}
+
+// nextPowerOf2 returns the next lowest power of 2 unless the input is a power
+// of two, in which case it returns the input
+func nextPowerOf2(v uint32) uint32 {
+	if v == 1 {
+		return 1
+	}
+
+	// find the next highest power using bit mashing
+	v--
+	v |= v >> 1
+	v |= v >> 2
+	v |= v >> 4
+	v |= v >> 8
+	v |= v >> 16
+	v++
+
+	// return the next lowest power
+	return v
```

Comment:
This code is slightly modified from where it was copied from here https://github.com/lazyledger/lazyledger-core/blob/93583ae898363c99a1a0983f890c0486a2a3f933/p2p/ipld/read.go#L274

The modification makes it return the next highest power of two. I'm not sure why unmodified version in the ipld package is the way it is. I'm worried that that version could cause bugs and will investigate further tomorrow.

Reply:
Yeah, I was also surprised that `nextPowerOf2` in ipld returns the next smaller or equal power of two. The naming should be better as it sounds like it will give you the next (larger or equal one).

Reply:
after some more inspection, the ipld package version is only being used to verify that a number is a power of two, and returns the next lowest power of two. I guess in that case it doesn't matter, but I ended up changing it to `isPowerOf2` in https://github.com/lazyledger/lazyledger-core/pull/331/commits/4fd8936fef8b63437f8c73d2d760fcb5f0548966